mh/equipment_utils.py

```diff
@@ -0,0 +1,1627 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Quest utils module.
+
+    Monster Hunter 3 Server Project
+    Copyright (C) 2023  Ze SpyRo
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import struct
+
+
+def slot(type_id, equipment_id, slots=0):
+    """Equipment slot / TODO: Handle gems"""
+    return struct.pack(">BBH", slots, type_id, equipment_id)
+    #return struct.pack(">BBHII", type_id, slots, equipment_id, 0, 0)
+
+
+def item_slot(item_id, quantity):
+    return struct.pack(">HH", item_id, quantity)
+
+
+def create_arena_equipment_set(sets):
+    arena_equipment_set = b""
+    for set_num in range(4):
+        weapon1, weapon2, weapon3, helm, chestpiece, gauntlets,\
+            faulds, leggings, items, gunner_pouch = sets[set_num]
+        # ---------- SET 1 ----------
+        # Weapon
+        arena_equipment_set += slot(*weapon1)
+        if weapon2 is not None:
+            arena_equipment_set += slot(*weapon2)
+        else:
+            arena_equipment_set += b"\xff" * 0x04
```

Comment:
How about creating a named constant for this empty weapon_slot `0xFFFFFFFF`?

Reply:
Fixed

Used the `struct` library to create named constants for empty arena equipment slots and empty arena item slots.

The distinction needed to be made between empty arena equipment slots and standard empty equipment slots due to the arena quest equipment slots being smaller in size/not including gems.